charmcraft.yaml

```diff
@@ -13,12 +13,30 @@ bases:
 
 parts:
   charm:
+    build-snaps:
+      - astral-uv
     build-packages:
       - git
       - libffi-dev
       - libssl-dev
       - pkg-config
-
     charm-binary-python-packages:
      - pydantic>2.0
      - pydantic-core
+    override-build: |
+      make requirements
+      craftctl default
+    # charm-requirements defaults to requirements.txt, but without setting it (counter-intuitive) 
+    # the charm does not respect the requirements.txt when generated dynamically during build.
+    charm-requirements:
+      - requirements.txt
+    prime:
+      - -*.charm
+      - -.venv
+      - -CONTRIBUTING.md
+      - -Makefile
+      - -pyproject.toml
+      - -README.md
+      - -tests/
+      - -uv.lock
+      - -*.egg-info
```

Comment:
- Can you confirm this `prime` is needed? I.e. would charmcraft take `.venv` or a previously packed `.charm` into a newly packed charm if we omit this section?
- Best practice: `prime` vs. [`.jujuignore`](https://github.com/canonical/prometheus-k8s-operator/blob/main/.jujuignore)?

Reply:
According to Alex Lowe, the `prime` or `stage` key is best to use rather than `.jujuignore`. I could not find any docks on how to do this with `stage`. Since `prime` was recently reworked to now exclude rather than include files, I will assume this is the most up to date approach for the future.